mlir/lib/Conversion/SCFToControlFlow/SCFToControlFlow.cpp

```diff
@@ -379,8 +378,8 @@ LogicalResult ForLowering::matchAndRewrite(ForOp forOp,
   // llvm.loop_annotation attribute.
   SmallVector<NamedAttribute> llvmAttrs;
   llvm::copy_if(forOp->getAttrs(), std::back_inserter(llvmAttrs),
-                [](auto attr) {
-                  return isa<LLVM::LLVMDialect>(attr.getValue().getDialect());
+                [](NamedAttribute attr) {
+                  return attr.getValue().getDialect().getNamespace() == "llvm";
```

Comment:
Why is this needed? Can `MLIRSCFToControlFlow` depend on the LLVM dialect?

Reply:
Using a string here looks like a bit of a hack to me :(

It's pretty unfortunate that we have to choose between a dependency and the feature here though.

Reply:
> Why is this needed? Can MLIRSCFToControlFlow depend on the LLVM dialect?

Well downstream users that don't use LLVM IR or LLVM MLIR Dialect may want to use this pass.   If downstream users are sensitive to the build/binary cost then they will have to duplicate this pass to eliminate the dependency. 

I guess it's not a huge deal to solve; we can expose the patterns as a separate function and have a callback to customize how attributes are propagated. The current behavior can remain the default and  downstream user can then instantiate a different pass without the LLVM dependency.

Reply:
removed changes and will follow up with workaround mentioned above